apps/web/src/components/Solana/ConnectWalletModal.tsx

```diff
@@ -0,0 +1,169 @@
+"use client";
+
+import { Box, Button, Text } from "@dex-web/ui";
+import type { WalletName } from "@solana/wallet-adapter-base";
+import { WalletReadyState } from "@solana/wallet-adapter-base";
+import type { Wallet } from "@solana/wallet-adapter-react";
+import { useWallet } from "@solana/wallet-adapter-react";
+import type { FC, MouseEvent } from "react";
+import {
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { createPortal } from "react-dom";
+import { twMerge } from "tailwind-merge";
+import { useWalletModal } from "./useWalletModal";
+import { WalletListItem } from "./WalletListItem";
+
+export interface WalletModalProps {
+  className?: string;
+  container?: string;
+}
+
+export const WalletModal: FC<WalletModalProps> = ({ container = "body" }) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const {
+    wallets,
+    select,
+    wallet: selectedWallet,
+    connecting,
+    connected,
+  } = useWallet();
+  const { setVisible } = useWalletModal();
+  const [fadeIn, setFadeIn] = useState(false);
+  const [portal, setPortal] = useState<Element | null>(null);
+
+  const [installedWallets, notInstalledWallets] = useMemo(() => {
+    const installed: Wallet[] = [];
+    const notInstalled: Wallet[] = [];
+
+    for (const wallet of wallets) {
+      if (wallet.readyState === WalletReadyState.Installed) {
+        installed.push(wallet);
+      } else {
+        notInstalled.push(wallet);
+      }
+    }
+
+    return [installed, notInstalled];
+  }, [wallets]);
+
+  const hideModal = useCallback(() => {
+    setFadeIn(false);
+    setTimeout(() => setVisible(false), 150);
+  }, [setVisible]);
+
+  const handleClose = useCallback(
+    (event: MouseEvent) => {
+      event.preventDefault();
+      hideModal();
+    },
+    [hideModal],
+  );
+
+  const handleWalletClick = useCallback(
+    (_event: MouseEvent, walletName: WalletName) => {
+      select(walletName);
+      // handleClose(event);
+    },
+    [select],
+  );
+
+  useEffect(() => {
+    if (selectedWallet && connected) {
+      hideModal();
+    }
+  }, [selectedWallet, connected, hideModal]);
+
+  useLayoutEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === "Escape") {
+        hideModal();
+      }
+    };
+
+    // Get original overflow
+    const { overflow } = window.getComputedStyle(document.body);
```

Comment:
Can help with doing this without this logic if needed, we shouldn't need to go outside of the react/css norms here!